splash/qtrender_image.py

```diff
@@ -457,13 +465,13 @@ def crop(self, rect):
         self.img = self.img.crop((left, top, right, bottom))
 
     def to_png(self, complevel=defaults.PNG_COMPRESSION_LEVEL):
-        buf = StringIO()
+        buf = BytesIO()
```

Comment:
Hey @kmike . I ran some performance checks and didn't see much difference. The time and memory consumed remained almost the same. Am I doing something wrong? Here is what I did.
```
In [5]:

from PIL import Image
In [3]:

from io import BytesIO
def to_png_bytesio(complevel=1):
    img = Image.open("pillow.jpg")
    buf = BytesIO()
    img.save(buf, 'png', compress_level=complevel)
    return buf.getvalue()
In [24]:

%timeit -n 1000 to_png_bytesio()
1000 loops, best of 3: 3.89 ms per loop
In [25]:

from cStringIO import StringIO
def to_png_stringio(complevel=1):
    img = Image.open("pillow.jpg")
    buf = StringIO()
    img.save(buf, 'png', compress_level=complevel)
    return buf.getvalue()
In [26]:

%timeit -n 1000 to_png_stringio()
1000 loops, best of 3: 3.9 ms per loop
In [1]:

%load_ext memory_profiler
In [8]:

from benchmark import to_png_bytesio
In [9]:

%mprun -f to_png_bytesio to_png_bytesio()
('',)

Filename: benchmark.py
​
Line #    Mem usage    Increment   Line Contents
================================================
    11     32.1 MiB      0.0 MiB   def to_png_bytesio(complevel=1):
    12     32.2 MiB      0.0 MiB       img = Image.open("pillow.jpg")
    13     32.2 MiB      0.0 MiB       buf = BytesIO()
    14     32.3 MiB      0.1 MiB       img.save(buf, 'png', compress_level=complevel)
    15     32.3 MiB      0.0 MiB       return buf.getvalue()
In [11]:

from benchmark import to_png_stringio
In [12]:

%mprun -f to_png_stringio to_png_stringio()
('',)

Filename: benchmark.py
​
Line #    Mem usage    Increment   Line Contents
================================================
     4     32.3 MiB      0.0 MiB   def to_png_stringio(complevel=1):
     5     32.3 MiB      0.0 MiB       img = Image.open("pillow.jpg")
     6     32.3 MiB      0.0 MiB       buf = StringIO()
     7     32.4 MiB      0.1 MiB       img.save(buf, 'png', compress_level=complevel)
     8     32.4 MiB      0.0 MiB       return buf.getvalue()
```